# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1    8437.673028         42               высшее             0   
1          1    4024.803754         36              среднее             1   
2          0    5623.422610         33              среднее             1   
3          3    4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0     926.185831         27               высшее             0   
6          0    2879.202052         43               высшее             0   
7          0     152.779569         50              среднее             1   
8          2    6929.865299         35               высшее             0   
9          0    2188.756445         41              среднее             1   
10         2    4171.483647         36               высшее             0   
11         0     792.701887         40              среднее             1   
12         0  365286.622650         65              среднее             1   
13         0    1846.641941         54  неоконченное высшее             2   
14         0    1844.956182         56               высшее             0   
15         1     972.364419         26              среднее             1   
16         0    1719.934226         35              среднее             1   
17         0    2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0   10038.818549         48              среднее             1   

       family_status  family_status_id gender income_type  debt  total_income  \
0    женат / замужем                 0      F   сотрудник     0        253875   
1    женат / замужем                 0      F   сотрудник     0        112080   
2    женат / замужем                 0      M   сотрудник     0        145885   
3    женат / замужем                 0      M   сотрудник     0        267628   
4   гражданский брак                 1      F   пенсионер     0        158616   
5   гражданский брак                 1      M   компаньон     0        255763   
6    женат / замужем                 0      F   компаньон     0        240525   
7    женат / замужем                 0      M   сотрудник     0        135823   
8   гражданский брак                 1      F   сотрудник     0         95856   
9    женат / замужем                 0      M   сотрудник     0        144425   
10   женат / замужем                 0      M   компаньон     0        113943   
11   женат / замужем                 0      F   сотрудник     0         77069   
12  гражданский брак                 1      M   пенсионер     0        118514   
13   женат / замужем                 0      F   сотрудник     0        130458   
14  гражданский брак                 1      F   компаньон     1        165127   
15   женат / замужем                 0      F   сотрудник     0        116820   
16   женат / замужем                 0      F   сотрудник     0        289202   
17  гражданский брак                 1      M   сотрудник     0         90410   
18    вдовец / вдова                 2      F   пенсионер     0         56823   
19         в разводе                 3      F   сотрудник     0        242831   

                                   purpose total_income_category  \
0                            покупка жилья                     B   
1                  приобретение автомобиля                     C   
2                            покупка жилья                     C   
3               дополнительное образование                     B   
4                          сыграть свадьбу                     C   
5                            покупка жилья                     B   
6                        операции с жильем                     B   
7                              образование                     C   
8                    на проведение свадьбы                     C   
9   

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

Сгруппируем данные о возврате кредита в срок, и посчитаем общее число заёмщиков для каждой категории.
*Значение столбца debt: 0 - кредит был возвращен в срок; 1 - кредит был просрочен*


In [22]:
data.groupby(['debt'])['children'].agg(['count'])

count
debt       
0     19599
1      1732

Далее будем рассматривать только целевую группу.

Посмотрим на количество должников для каждой категории и на долю должников от общего числа семей:

In [23]:
pd.pivot_table(data, values='debt', index='children', aggfunc=['count','sum', 'mean'])

count   sum      mean
           debt  debt      debt
children                       
0         14091  1063  0.075438
1          4808   444  0.092346
2          2052   194  0.094542
3           330    27  0.081818
4            41     4  0.097561
5             9     0  0.000000

Так как группы должников с 3 и 4 детьми слишком малочисленны, то для дальнейшего анализа целесообразным будет соотносить только сравнимые по размеру группы (0, 1 и 2 ребенка).

Найдем процент просроченных платежей по отношению к количеству детей в каждой группе:

In [24]:
t = data[data['debt'] == 1]['children'].value_counts() / data['children'].value_counts() * 100
t.loc[:2]

0    7.543822
1    9.234609
2    9.454191
Name: children, dtype: float64

**Вывод:**

*Рассматривая зависимости между семьями с 0, 1 и 2 детьми, можно сделать вывод, что* ***Семьи без детей значительно реже просрочивают выплаты по кредиту***.
*В семьях с одним ребенком процент просроченных платежей равен 9.234%, в семьях с двумя детьми этот процент составляет уже 9.454%, из чего так же можно сделать вывод, что* ***Чем больше у заёмщика детей, тем менее он надёжен***.

In [25]:
# Код ревьюера
data.pivot_table(index = 'children', values = 'debt', aggfunc = ['count', 'sum', 'mean'])

count   sum      mean
           debt  debt      debt
children                       
0         14091  1063  0.075438
1          4808   444  0.092346
2          2052   194  0.094542
3           330    27  0.081818
4            41     4  0.097561
5             9     0  0.000000

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [26]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.

Сгруппируем данные о возврате кредита в срок, соотнесем их с семейным положением заёмщика и посчитаем количество значений для каждой группы.
*Значение столбца debt: 0 - кредит был возвращен в срок; 1 - кредит был просрочен*.

In [27]:
data.groupby('debt')['family_status'].agg(['count'])

count
debt       
0     19599
1      1732

Посмотрим на  соотношение количество семей в каждой группе семейного положения и  на долю должников от общего количества семей:

In [28]:
pd.pivot_table(data, values='debt', index='family_status', aggfunc=['count','sum', 'mean'])

count  sum      mean
                        debt debt      debt
family_status                              
Не женат / не замужем   2796  273  0.097639
в разводе               1189   84  0.070648
вдовец / вдова           951   63  0.066246
гражданский брак        4134  385  0.093130
женат / замужем        12261  927  0.075606

Найдем процент просроченных платежей по каждой группе семейного положения:

In [29]:
(data[data['debt'] == 1]['family_status'].value_counts() / data['family_status'].value_counts() * 100).sort_values(ascending=False)

Не женат / не замужем    9.763948
гражданский брак         9.313014
женат / замужем          7.560558
в разводе                7.064760
вдовец / вдова           6.624606
Name: family_status, dtype: float64

**Вывод:** 
Исходя из результатов выше можно предположить, что самыми ненадежными заемщиками являются *не женатые* и *пары в гражданском браке*. Однако, банк - это про деньги, так что стоит проверить зависимость между *целью кредита, уровнем дохода и возвратом кредита в срок*.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [30]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.

Сгруппируем данные о возврате кредита в срок, соотнесем их с уровнем дохода заёмщика и посчитаем количество значений для каждой группы.
*Значение столбца debt: 0 - кредит был возвращен в срок; 1 - кредит был просрочен*.
Ранее мы уже создали столбец `total_income_category`, в котором разделили людей на группы по уровню дохода, где:
- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.

In [31]:
data.groupby('debt')['total_income_category'].agg(['count'])

count
debt       
0     19599
1      1732

Посмотрим на соотношение уровня дохода и на долю должников по каждой категории:

In [32]:
pd.pivot_table(data, values='debt', index='total_income_category', aggfunc=['count','sum', 'mean'])

count   sum      mean
                        debt  debt      debt
total_income_category                       
A                         25     2  0.080000
B                       5014   354  0.070602
C                      15921  1353  0.084982
D                        349    21  0.060172
E                         22     2  0.090909

Теперь выделим из этих данных только целевые:

In [33]:
data[data['debt'] == 1]['total_income_category'].value_counts()

C    1353
B     354
D      21
A       2
E       2
Name: total_income_category, dtype: int64

Из данных выше видно, что категории `A` и `E` слишком малочисленны. Будет целесообразным сравнивать категории схожими по размеру: `c`, `B` и `D`.

Найдем процент просроченных платежей по каждой категории дохода:

In [34]:
(data[data['debt'] == 1]['total_income_category'].value_counts() / data['total_income_category'].value_counts() * 100).sort_values(ascending=False)

E    9.090909
C    8.498210
A    8.000000
B    7.060231
D    6.017192
Name: total_income_category, dtype: float64

**Вывод:** 

Так как категории `A` и `E` слишком малочисленны, для итоговых выводов мы их использовать не будем.
По остальным катешориям можно сказать, что самый большой процент невозврата у заёмщиков категории `C` - людей со средним доходом. В категориях `B` и `D` процент оказался самым низким. Для категории `D` это может быть связано с тем, что люди с маленьким доходом имеют меньше материальных амбиций. Возможно, если бы в исходных данных была информация о сумме кредита, то можно было сказать больше про эту категорию. Заёмщики в категории `B` ожидаемо оказались самыми добросовестными.
Сравнивая зависимости между категориями `C` и `B` можно утверждать, что уровень дохода напрямую влияет на надежность заёмщика.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

---

Сгруппируем данные о возврате кредита в срок, соотнесем их с разными целями кредита, посчитаем количество значений для каждой группы и выведем долю должников по каждой цели.
*Значение столбца debt: 0 - кредит был возвращен в срок; 1 - кредит был просрочен*.
Ранее мы уже создали столбец `purpose_category`, в котором разделили цели на 4 категории:

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

In [36]:
pd.pivot_table(data, values='debt', index='purpose_category', aggfunc=['count','sum', 'mean'])

count  sum      mean
                           debt debt      debt
purpose_category                              
операции с автомобилем     4279  400  0.093480
операции с недвижимостью  10751  780  0.072551
получение образования      3988  369  0.092528
проведение свадьбы         2313  183  0.079118

Оставим только целевой столбец:

In [37]:
data[data['debt'] == 1]['purpose_category'].value_counts()

операции с недвижимостью    780
операции с автомобилем      400
получение образования       369
проведение свадьбы          183
Name: purpose_category, dtype: int64

Переведем полученные данные в проценты:

In [38]:
(data[data['debt'] == 1]['purpose_category'].value_counts() / data['purpose_category'].value_counts() * 100).sort_values(ascending=False)

операции с автомобилем      9.347978
получение образования       9.252758
проведение свадьбы          7.911803
операции с недвижимостью    7.255139
Name: purpose_category, dtype: float64

**Вывод:** 

Исходя из вычислений видно, что чаще всего просрочки происходят по кредиту на `операции с автомобилем` и `получение образования`. Доли просрочек на `проведение свадьбы` и `операции с недвижимостью` так же примерно равные.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 

Пропуски в исходных данных были обнаружены в столбцах `days_employed` и `total_income`. 
Вероятно, пропуски в столбце `days_employed` произошли в результате технической ошибки при переводе файла из одного формата в другой, на это указывают отрицательные значения.

В столбце `total_income` пропуски могли появиться как из-за технической ошибки, так и из-за человеческого фактора.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 

Пропуски в мы заменили *медианным* значением, так как при таком способе заполнения мы заранее отметаем аномально большие и маленькие значения, тогда наши данные лучше отражают действительность и не будут искажать дальнейшие вычисления.

### Шаг 4: общий вывод.

**Вывод**:

Была проведена работа над банковскими данными и проведено исследование, в котором мы выясняли ***Какие факторы влияют на возврат кредита в срок.***

**В ходе исследования мы провели предобработку данных, в которой:**

- избавились от дубликатов
- заполнили пропущенные значения
- привели все данные к правильному типу данных
- категоризировали часть данных для упрощения анализа



**Подводя итог исследования можно выделить зависимости:**
- количество детей влияет на надёжность заёмщика. Чем больше детей - тем больше вероятность просрочки;
- заёмщики состоящие или состоявшие в браке реже имеют просрочки;
- с увеличением дохода растет и стабильность в обслуживании своего долга;
- кредиты, взятые на покупку недвижимости возвращаются чаще всего. Самыми же проблемными выглядят автокредиты и кредиты на образование.


**Мысли/рекоммендации:**

Проводя исследование, у меня появлялись мысли о том, что будь в исходных данных несколько дополнительных столбцов, например: `Данные о сумме кредита` и `Данные о ежемесячном доходе заёмщика`, то исследование могло стать более подробным. Тогда получилось бы выявить более подробные зависимости, и даже, наверное, на этих данных строить модель машинного обучения, котороая присуждала бы каждому заёмщику свой кредитный рейтинг, что в дальнейшем могло быть использовано для прогнозирования надежности клиента и его платежеспособности исходя из статистики по всем факторам. 